Library/src/Multiplayer/Components/ConversationSpaceComponent.cpp

```diff
@@ -274,4 +751,121 @@ const csp::common::String& ConversationSpaceComponent::GetConversationId() const
 	return GetStringProperty(static_cast<uint32_t>(ConversationPropertyKeys::ConversationId));
 }
 
+void ConversationSpaceComponent::StoreConversationMessage(const csp::systems::Space& Space,
+														  const csp::common::String& UserId,
+														  const csp::common::String& Message,
+														  MessageResultCallback Callback) const
+{
+	auto ConversationId = GetConversationId();
+
+	if (ConversationId.IsEmpty())
+	{
+		CSP_LOG_WARN_MSG("This component does not have a conversation set! Call CreateConversation before doing message operations.");
+
+		INVOKE_IF_NOT_NULL(Callback, MakeInvalid<MessageResult>());
+
+		return;
+	}
+
+	const csp::systems::AssetCollectionResultCallback AddCommentCallback = [=](const csp::systems::AssetCollectionResult& AddCommentResult)
+	{
+		if (AddCommentResult.GetResultCode() == csp::systems::EResultCode::InProgress)
+		{
+			return;
+		}
+
+		if (AddCommentResult.GetResultCode() == csp::systems::EResultCode::Success)
```

Comment:
It would just mean having an if block for the failure case and then moving success out of the nested if block. It doesn't change anything about the execution, just matches the flow convention used everywhere else.